### PR TITLE
fix: break ceiling replay boot deadlock

### PR DESCRIPTION
**Pull request by carlosflorencio**
> [!TIP]
> **PR walkthrough:** [Open the visual walkthrough](https://walkthrough.kandev.ai/pr/3776/8e114cd7c17a.html)

Deferred session replay could deadlock with an agent's boot-ready callback and block all new task scheduling. Release task admission before replay dispatch and acquire it before the global task-state mutex so callbacks and unrelated launches can progress.

The durable replay claim and final workflow-binding checks remain in place. The fix package and completed work order are in `docs/plans/ceiling-boot-deadlock/`.

## Validation

- Both new regression tests failed against the original code and passed after the correction.
- `go test -race ./internal/orchestrator -run 'Test(Ceiling|ReplayCeiling|ClaimCeiling|ClearCeiling|WriteTaskReview|ReviewAdmission|HandleAgentBootReady|AgentBootReady|WorkflowEntryDispatch)' -count=1 -timeout=5m` passed.
- `python3 scripts/list-docs.py validate`, `python3 scripts/lint-spec-files.py --all`, and `git diff --check` passed.
- Internal design updated. Public docs already describe automatic replay. No UI changes or screenshots are required.
- The live instance was not restarted or modified.

## Checklist

- [ ] If I do not have repository write access and this is a large architectural change, I discussed the direction in a linked issue before opening this PR.
- [ ] This PR contains one logical change; unrelated work is split into separate PRs.
- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [ ] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed.

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/3776?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/backend/internal/orchestrator/ceiling_replay.go [hunk 1]
```diff
@@ -404,29 +404,37 @@ func stripCeilingRecordKeys(existing interface{}) map[string]interface{} {
...
-	if task == nil || task.ID == "" {
-		return ceilingReplayFailed
-	}
-	ctx, release := s.lockCeilingEntryAdmission(ctx, task.ID)
+// loadCeilingReplayTask releases admission before session guards or provider
+// callbacks can run. Only the immutable workflow binding crosses into dispatch;
+// the durable claim owns replay until settlement.
+func (s *Service) loadCeilingReplayTask(ctx context.Context, taskID string, deferral models.CeilingDeferral) (*models.Task, ceilingReplayOutcome) {
+	ctx, release := s.lockCeilingEntryAdmission(ctx, taskID)
 	defer release()
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/orchestrator/ceiling_replay.go:412]: Fixed in `8e114cd7c1`.

The replay now commits immutable workflow-entry ownership under the task admission lock, releases that lock before provider and readiness I/O, and carries a process-local dispatch fence through the dispatch. Competing workflow route writers are rejected until the admitted dispatch settles. Prepared and dynamic launches, resumes, and prompt dispatch all use this boundary.

Added `TestCeilingReplayCommitsRouteOwnershipBeforeProviderDispatch`, which attempts a successor route update after the final validation and verifies that the old route remains authoritative. The targeted race suite and the complete `internal/orchestrator` package pass.
